### PR TITLE
Deprecate Ember.merge

### PR DESCRIPTION
**Pull request by RobbieTheWagner**

### CODE REVIEW

File: packages/internal-test-helpers/lib/system/synthetic-events.js [hunk 1]
```diff
@@ -124,16 +124,16 @@ export function fireEvent(element, type, options = {}) {
...
 }
 
 function buildMouseEvent(type, options = {}) {
   let event;
   try {
     event = document.createEvent('MouseEvents');
-    let eventOpts = merge({}, DEFAULT_EVENT_OPTIONS);
-    merge(eventOpts, options);
+    let eventOpts = assign({}, DEFAULT_EVENT_OPTIONS);
+    assign(eventOpts, options);
```
Comment (rwjblue) [inline packages/internal-test-helpers/lib/system/synthetic-events.js]: Can you replace this with a single assign?

File: packages/internal-test-helpers/lib/system/synthetic-events.js [hunk 2]
```diff
@@ -162,8 +162,8 @@ function buildKeyboardEvent(type, options = {}) {
   let event;
   try {
     event = document.createEvent('KeyEvents');
-    let eventOpts = merge({}, DEFAULT_EVENT_OPTIONS);
-    merge(eventOpts, options);
+    let eventOpts = assign({}, DEFAULT_EVENT_OPTIONS);
+    assign(eventOpts, options);
```
Comment (rwjblue) [inline packages/internal-test-helpers/lib/system/synthetic-events.js]: Same here? We should only need one assign...

File: packages/@ember/polyfills/lib/merge.ts [hunk 3]
```diff
@@ -25,6 +27,12 @@ export default function merge<T, U>(original: T, updates: U): T & U;
   @public
 */
 export default function merge(original: object, updates: object) {
+  deprecate('Use of `merge` has been deprecated. Please use `assign` instead.', false, {
+    id: 'ember-polyfills.deprecate-merge',
+    until: '4.0.0',
+    url: 'https://emberjs.com/deprecations/v3.x/#toc_ember-polyfills-deprecate-merge',
```
Comment (rwjblue) [inline packages/@ember/polyfills/lib/merge.ts]: Is this created yet? Can you link me to the deprecation app PR?

**Review by rwjblue — CHANGES_REQUESTED** [review]

**Comment by rwjblue** [comment]
We also need to guard this with deprecation flags for svelting, something like https://github.com/emberjs/ember.js/pull/16745.

File: packages/@ember/polyfills/lib/merge.ts (hunk 3 above)
Comment (rwwagner90) [reply packages/@ember/polyfills/lib/merge.ts]: This is not created yet. I need to look into what I need to do for that.

Comment (rwwagner90) [reply packages/@ember/polyfills/lib/merge.ts]: Just created it https://github.com/ember-learn/deprecation-app/pull/179

**Comment by rwwagner90** [comment]
@rwjblue I added a link to the deprecations app PR and attempted the svelting stuff. Not sure I did it exactly right, so feel free to let me know what I should tweak. One thing I wasn't sure on is what version we should mark this deprecation as introduced in? I randomly chose 3.5 for both the svelting here and the deprecations app PR, but I think it will have to be a later version probably, since I think 3.5 beta has already been cut?

**Comment by rwwagner90** [comment]
@rwjblue just updated to point to 3.6. Let me know if there are any other tweaks I should make 😄

**Comment by rwjblue** [comment]
I merged the deprecation PR, it should redeploy soon, would you mind confirming the url here works properly?

Once we’ve done that we can merge...

**Comment by locks** [comment]
https://emberjs.com/deprecations/v3.x/#toc_ember-polyfills-deprecate-merge is up and working :)

**Comment by rwwagner90** [comment]
@rwjblue looks like it's deployed and working. Anything else we need to tweak here?

**Review by rwjblue — APPROVED** [review]

**Pull request merged**